packages/diagrams/frontend/sirius-components-diagrams/src/renderer/panel/DiagramPanelContext.tsx

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import React, { useState } from 'react';
+import { DiagramPanelContextValue, DiagramPanelContextProviderProps } from './DiagramPanelContext.types';
+
+const defaultContextValue: DiagramPanelContextValue = {
+  isMiniMapVisible: false,
+  setMiniMapVisibility: () => {},
+  isSnapToGridEnabled: false,
+  setSnapToGridEnabled: () => {},
+  isHelperLineEnabled: false,
+  setHelperLineEnabled: () => {},
+};
+
+export const DiagramPanelContext = React.createContext<DiagramPanelContextValue>(defaultContextValue);
+
+const isLocalStorageAvailable = (): boolean => {
+  const test = 'localStorageTest';
+  try {
+    localStorage.setItem(test, test);
+    localStorage.removeItem(test);
+    return true;
+  } catch (_) {
+    return false;
+  }
+};
+
+export const DiagramPanelContextProvider = ({ children }: DiagramPanelContextProviderProps) => {
+  const localStorageKey: string = 'sirius-diagram-mini-map-visibility';
+
+  const getInitialMiniMapVisibility = (): boolean => {
+    if (!isLocalStorageAvailable()) {
+      return true;
+    }
+    const storedPreference = localStorage.getItem(localStorageKey);
+    return storedPreference ? JSON.parse(storedPreference) : true;
+  };
+
+  const [miniMap, setMiniMap] = useState<boolean>(getInitialMiniMapVisibility);
+  const [snapToGrid, setSnapToGrid] = useState<boolean>(false);
+  const [helperLine, setHelperLine] = useState<boolean>(true);
```

Comment:
Those three states have nothing to do together, the support from the helper line or the snap to grid has nothing to do with the local storage support.